### ISSUE TITLE
Lookup events by event_id

### ISSUE THREAD

**Issue by martinetd** [matrix-org/matrix-rust-sdk#242]
It's been a while so I updated my version of the sdk this weekend, thanks for the continued work! Doing that more often would probably save me some time figuring out errors :p

Anyway, I see ruma now has support for reactions, I now get events like this:
```
Message(Reaction(SyncMessageEvent { content: ReactionEventContent { relation: Relation { event_id: $efB7k-Y7b52sE0D84zudUwGEgewRGsoJKdSd9x8XoW4, emoji: "😀" } }, event_id: $h2t3OTbkmMTDp8_fg-300xv9sq_GA7Oa98s84GFAAXo, sender: @xxx:xxx.org, origin_server_ts: MilliSecondsSinceUnixEpoch(1621750128000), unsigned: Unsigned { age: Some(260), transaction_id: None, relations: None } }))
```

There also are redaction events:
```
Message(RoomRedaction(SyncRedactionEvent { content: RedactionEventContent { reason: None }, redacts: $Y2pBXwHshkUBHiR_t3p-9hqS_WXEhvoejH9FVmAJgZM, event_id: $EqFOJNNejFxCX7ZKx0pepxdpJqsiT-qbSLXmw7ob0I4, sender: @asmadeus:codewreck.org, origin_server_ts: MilliSecondsSinceUnixEpoch(1621753269784), unsigned: Unsigned { age: Some(399), transaction_id: None, relations: None } }))
```

In both cases the event refers to another event by event_id, and I'd like to display something about the original message (e.g. part of the original message body to identify what has been reacted/redacted), but I can't see anything in the API that'd let me get the event back from its id.

Realistically, I don't expect the message to be cached in the client's memory, so I would be fine with the lookup operation being a request sent to the homeserver with a custom reply? or something coming back in the next sync iteration -- I don't know how the protocol works at this level, I'm afraid heavy clients like element just remember all the recent messages and look it up locally? I'd rather not write down all the messages I've seen in a cache...

Thanks!

**Comment by martinetd** [comment]
Also while I'm on redactions I'm not sure if it's element web that doesn't actually set the reason or matrix-rust-sdk that doesn't decrypt it properly but in the encrypted case, the redaction reason seems to always be set to None.

Looking at the event source I see some ciphertext as well redacts id that is in plain text, so I would assume the ciphertext contains the reason, but I didn't see any way of getting element web to actually display the reason once set so it's hard to tell :|
Anyway, that's not important for me right now; if folks aren't used to it being displayed in the first place they probably won't be using it much...

**Comment by poljar** [comment]
As for your first issue, the method doing that specific request just hasn't been added yet, relevant spec entry is [here](https://spec.matrix.org/unstable/client-server-api/#getting-events-for-a-room).

The second issue, not sure. Historically Element has been ignoring the reason and I seem to remember Element doing something silly as encrypting an empty dict `{}` for the reason. I think that's been somewhat of a grey area since you need to send encrypted stuff as `m.room.encrypted` but then the server doesn't know that it's a redaction and the API endpoint for redactions doesn't seem to support encryption either: https://spec.matrix.org/unstable/client-server-api/#put_matrixclientr0roomsroomidredacteventidtxnid

**Comment by martinetd** [comment]
Thanks for the quick reply.

1/ Ok, I see it's implemented in ruma so in theory we just need to wrap it like we do for e.g. create_room? I plugged in a LRU for recent messages but I guess I can give it a shot at some point, already spent quite some more time than planned on this so probably not this weekend..


2/ Yeah, I'm not sure why they prompt for setting a reason and then pretty much ignore it, I had never seen it before trying today. I don't think I care enough to get this clarified/better supported, will let the lion sleep on this one...

**Comment by poljar** [comment]
Yes, just needs a higher level method that does the right thing. Do note that you can already send out such a request using https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk/struct.Client.html#method.send, but that won't automatically decrypt the event for you so a helper is the way to go.

**Comment by emgrav** [comment]
This can likely be closed now that #298 is merged.

**Comment by poljar** [comment]
Right, thanks for noticing.
